homeassistant/components/timer.py

```diff
@@ -0,0 +1,402 @@
+"""
+Timer component.
+
+For more details about this component, please refer to the documentation
+at https://home-assistant.io/components/timer/
+"""
+import asyncio
+import logging
+import os
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.util.dt as dt_util
+import homeassistant.helpers.config_validation as cv
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (ATTR_ENTITY_ID, CONF_ICON, CONF_NAME)
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.restore_state import async_get_last_state
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.util import Throttle
+
+from homeassistant.loader import bind_hass
+
+_LOGGER = logging.getLogger(__name__)
+
+UPDATE_INTERVAL = timedelta(minutes=1)
+
+EVENT_TIMER_STARTED = 'timer.started'
```

Comment:
Are these events needed? Can't this one be derived from the state being set?

I suggest we remove these events for now and only add them if a use case for them comes up.

Reply:
Yeah, all these events should be removed.

Reply:
I'm fine with removing the `paused` and `started` events, since we can infer this from the state. I would like to keep the `cancelled` and `finished` events though, because the state can change from X to `idle` either by cancelling or by finishing. As an example: the pending time before an alarm fires while we're waiting for a login. If the login is successful we cancel, if we don't and the timer expires, it finishes. In the latter case it triggers a siren, whereas a cancelled timer wouldn't trigger the siren. We wouldn't know that if we just monitor the state.

Reply:
Makes sense. Let's keep cancelled and finished 👍